pkg/platform/abstract/platform.go

```diff
@@ -177,6 +179,16 @@ func (ap *Platform) HandleDeployFunction(existingFunctionConfig *functionconfig.
 		return nil, errors.Wrap(err, "Failed to deploy function")
 	}
 
+	if skipDeploy {
+		ap.Logger.Info("Skipping function deployment")
+		return &platform.CreateFunctionResult{
+			CreateFunctionBuildResult: platform.CreateFunctionBuildResult{
+				Image:                 createFunctionOptions.FunctionConfig.Spec.Image,
+				UpdatedFunctionConfig: createFunctionOptions.FunctionConfig,
+			},
+		}, nil
+	}
```

Comment:
This is handled in `onAfterBuild` for localPlatform. Why not for the general use case?
This code bit should be `onAfterBuild` implementation imo

Reply:
Because then there'll be duplication in `OnAfterBuild` for local and `OnAfterBuild` for kube. This way there isn't any duplication. More over, this has to happen in this function, so it can return and not continue to the next lines of code

Reply:
Explained offline